modin/experimental/cloud/local_cluster.py

```diff
@@ -0,0 +1,94 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+import subprocess
+import sys
+
+from .base import ConnectionDetails
+from .cluster import BaseCluster
+from .connection import Connection
+from .rpyc_proxy import WrappingConnection, WrappingService, _TRACE_RPYC
+from .tracing.tracing_connection import TracingWrappingConnection
+
+
+class LocalWrappingConnection(
+    TracingWrappingConnection if _TRACE_RPYC else WrappingConnection
+):
+    def _init_deliver(self):
+        def ensure_modin(modin_init):
+            import sys
+            import os
+
+            modin_dir = os.path.abspath(os.path.join(os.path.dirname(modin_init), ".."))
+            # make sure "import modin" will be taken from current modin, not something potentially installed in the system
+            if modin_dir not in sys.path:
+                sys.path.insert(0, modin_dir)
```

Comment:
What about ensure python version?

Reply:
This isn't ensuring a version, it's ensuring importability of `modin` from the tests being run, not from something installed in current Python.

Reply:
But we can have problems with Python compatibility if it has different versions locally and remotely. Can we add a check for this?

Reply:
Again, this function is used only _locally_ (e.g. when running in simulated mode). In local run remote end runs via exact same _Python binary_ as local end, so the check would be moot.